src/js/common/utils/lodash-replacements/cloneDeep.js

```diff
@@ -0,0 +1,34 @@
+import clone from './clone';
+
+/**
+ * Deeply clones an object. It's probably not super performant on deeply nested objects, but
+ *  it gets the job done for now.
+ *
+ * Functions retain the same reference, but their `this` context changes like we'd expect.
+ *
+ * @param {Array|Object} object
+ * @return {Object}
+ */
+export default function cloneDeep(object) {
+  const newObj = Array.isArray(object) ? object.slice() : clone(object);
+
+  const cloneArrayElement = e => {
+    // Deep clones arrays and objects
+    if (typeof e === 'object' && e !== null) {
+      return cloneDeep(e);
+    }
+
+    return e;
+  };
+
+  for (const key of Object.keys(newObj)) {
```

Comment:
Is there a reason for this vs `forEach`?

Reply:
Mmm...I think this is just vestigial from when I was doing it differently? I'll change it to use `forEach()`. Good call.